jerry-core/ecma/builtin-objects/ecma-builtin-json.c

```diff
@@ -923,7 +923,12 @@ static ecma_value_t ecma_builtin_json_str_helper (const ecma_value_t arg1, /**<
                                                   empty_str_p,
                                                   arg1,
                                                   false);
-  JERRY_ASSERT (ecma_is_value_true (put_comp_val));
+  if (!ecma_is_value_true (put_comp_val))
+  {
+    ecma_deref_object (obj_wrapper_p);
```

Comment:
Is this enough? Some lines below, the normal code path frees `put_comp_val`, `empty_str_p`, and `obj_wrapper_p` before returning. I think the error handling code path should also do proper cleanup. Perhaps
```c
if (ecma_is_value_true (put_comp_val))
{
  ECMA_TRY_CATCH (str_val,
                  ecma_builtin_json_str (empty_str_p, obj_wrapper_p, &context),
                  ret_value);
  ret_value = ecma_copy_value (str_val);
  ECMA_FINALIZE (str_val);
}
else
{
  ret_value = ECMA_VALUE_UNDEFINED;
}
```
might work better.

Note: the normal code path mysteriously frees `put_comp_val ` twice. That could be fixed now, I think.